src/HttpDriver/Configuration.php

```diff
@@ -3,31 +3,92 @@
 namespace GraphAware\Neo4j\Client\HttpDriver;
 
 use GraphAware\Common\Driver\ConfigInterface;
+use Http\Client\HttpClient;
+use Http\Discovery\HttpClientDiscovery;
+use Http\Discovery\MessageFactoryDiscovery;
+use Http\Message\RequestFactory;
 
 class Configuration implements ConfigInterface
 {
     /**
      * @var int
+     * @deprecated Will be removed in 5.0
      */
     protected $timeout;
 
     /**
      * @var string
+     * @deprecated Will be removed in 5.0
      */
     protected $curlInterface;
 
     /**
+     * @var HttpClient
+     */
+    private $httpClient;
+
+    /**
+     * @var RequestFactory
+     */
+    private $requestFactory;
+
+    /**
      * @return Configuration
      */
-    public static function create()
+    public static function create(HttpClient $httpClient = null, RequestFactory $requestFactory = null)
+    {
+        $config = new self();
+        $config->httpClient = $httpClient ?: HttpClientDiscovery::find();
+        $config->requestFactory = $requestFactory ?: MessageFactoryDiscovery::find();
+
+        return $config;
+    }
+
+    /**
+     * @return HttpClient
+     */
+    public function getHttpClient()
     {
-        return new self();
+        return $this->httpClient;
+    }
+
+    /**
+     * @param HttpClient $httpClient
+     *
+     * @return Configuration
+     */
+    public function setHttpClient(HttpClient $httpClient)
+    {
+        $this->httpClient = $httpClient;
```

Comment:
I would prefer you return a new instance of the Configuration class with the http client injected